From 3d234c1faa5396a96b501e4355044c2bec1ecfaf Mon Sep 17 00:00:00 2001
From: TheWhisp <daavis.90@gmail.com>
Date: Sat, 16 Mar 2013 16:05:44 +0200
Subject: [PATCH] fix Bluetooth

Gingerbread uses older Bluez version than ICS/JB
Forward-port some changes from Samsung's opensource package

Change-Id: Ia0deec0cc64c92d855b4d21a91ff07db402f51f9
---
 lib/bluetooth/bluetooth.h |    8 ++++++--
 lib/bluetooth/hci.h       |   10 +++++++---
 plugins/hciops.c          |   24 ++++++++++++------------
 3 files changed, 25 insertions(+), 17 deletions(-)

diff --git a/lib/bluetooth/bluetooth.h b/lib/bluetooth/bluetooth.h
index 1c65a37..000a12c 100644
--- a/lib/bluetooth/bluetooth.h
+++ b/lib/bluetooth/bluetooth.h
@@ -71,15 +71,19 @@ struct bt_security {
 
 #define BT_DEFER_SETUP	7
 
-#define BT_FLUSHABLE	8
+#define BT_FLUSHABLE	10
 
 #define BT_FLUSHABLE_OFF	0
 #define BT_FLUSHABLE_ON		1
 
-#define BT_POWER	9
+#define BT_POWER	8
 struct bt_power {
 	uint8_t force_active;
 };
+#define BT_AMP_POLICY	9
+#define BT_AMP_POLICY_REQUIRE_BR_EDR	0
+#define BT_AMP_POLICY_PREFER_AMP		1
+#define BT_AMP_POLICY_PREFER_BR_EDR		2
 
 /* Connection and socket states */
 enum {
diff --git a/lib/bluetooth/hci.h b/lib/bluetooth/hci.h
index c3bebe9..5d052de 100644
--- a/lib/bluetooth/hci.h
+++ b/lib/bluetooth/hci.h
@@ -34,7 +34,7 @@ extern "C" {
 
 #define HCI_MAX_DEV	16
 
-#define HCI_MAX_ACL_SIZE	1024
+#define HCI_MAX_ACL_SIZE	1500
 #define HCI_MAX_SCO_SIZE	255
 #define HCI_MAX_EVENT_SIZE	260
 #define HCI_MAX_FRAME_SIZE	(HCI_MAX_ACL_SIZE + 4)
@@ -92,6 +92,7 @@ enum {
 #define HCIGETCONNLIST	_IOR('H', 212, int)
 #define HCIGETCONNINFO	_IOR('H', 213, int)
 #define HCIGETAUTHINFO	_IOR('H', 215, int)
+#define HCISETAUTHINFO	_IOR('H', 216, int)
 
 #define HCISETRAW	_IOW('H', 220, int)
 #define HCISETSCAN	_IOW('H', 221, int)
@@ -2117,9 +2118,12 @@ typedef struct {
 	uint8_t		bdaddr_type;
 	bdaddr_t	bdaddr;
 	uint8_t		length;
-	uint8_t		data[0];
+//	uint8_t		data[0];
+	uint8_t		data[31];
+	uint8_t		rssi;
 } __attribute__ ((packed)) le_advertising_info;
-#define LE_ADVERTISING_INFO_SIZE 9
+//#define LE_ADVERTISING_INFO_SIZE 9
+#define LE_ADVERTISING_INFO_SIZE 41
 
 #define EVT_LE_CONN_UPDATE_COMPLETE	0x03
 typedef struct {
diff --git a/plugins/hciops.c b/plugins/hciops.c
index ad97729..e25e546 100644
--- a/plugins/hciops.c
+++ b/plugins/hciops.c
@@ -1605,10 +1605,10 @@ static void read_bd_addr_complete(int index, read_bd_addr_rp *rp)
 
 static inline void cs_inquiry_evt(int index, uint8_t status)
 {
-	if (status) {
-		error("Inquiry Failed with status 0x%02x", status);
-		return;
-	}
+//	if (status) {
+//		error("Inquiry Failed with status 0x%02x", status);
+//		return;
+//	}
 
 	set_state(index, DISCOV_INQ);
 }
@@ -1745,10 +1745,10 @@ static inline void inquiry_complete_evt(int index, uint8_t status)
 	int adapter_type;
 	struct btd_adapter *adapter;
 
-	if (status) {
-		error("Inquiry Failed with status 0x%02x", status);
-		return;
-	}
+//	if (status) {
+//		error("Inquiry Failed with status 0x%02x", status);
+//		return;
+//	}
 
 	adapter = manager_find_adapter_by_id(index);
 	if (!adapter) {
@@ -1770,10 +1770,10 @@ static inline void inquiry_complete_evt(int index, uint8_t status)
 
 static inline void cc_inquiry_cancel(int index, uint8_t status)
 {
-	if (status) {
-		error("Inquiry Cancel Failed with status 0x%02x", status);
-		return;
-	}
+//	if (status) {
+//		error("Inquiry Cancel Failed with status 0x%02x", status);
+//		return;
+//	}
 
 	set_state(index, DISCOV_HALTED);
 }
-- 
1.7.9.5

